lisa/wlgen/workload.py

```diff
@@ -133,16 +133,19 @@ def run(self, cpus=None, cgroup=None, background=False, as_root=False, timeout=N
 
         logger.info("Execution start: %s", _command)
 
+        # Just return the Popen handle
         if background:
-            target.background(_command, as_root=as_root)
-        else:
-            self.output = target.execute(_command, as_root=as_root, timeout=timeout)
-            logger.info("Execution complete")
+            return target.background(_command, as_root=as_root)
```

Comment:
I'd rather have `run(background=True)` return None as before with a DeprecationWarning and introduce a new `run_background()` function that does the right thing. Ultimately, we can kill the `background` parameter of `run()`, which will mean:
* `run()` is meant to be the foreground run, which is what most people will need
* `run_background()` for background runs, returning the handle, and maybe some background-specific options.

If that makes sense from code-sharing point of view, we can have an internal `_run()` that returns one or the other, but IMO it's better to keep background-related options/return value separated.